test/lint/packages.js

```diff
@@ -0,0 +1,122 @@
+// Any copyright is dedicated to the Public Domain.
+// http://creativecommons.org/publicdomain/zero/1.0/
+
+import expect from 'expect';
+import fs from 'fs-promise';
+import glob from 'glob';
+import difference from 'lodash/difference';
+
+const all = '/**/';
+const valid = '*.@(js|jsx)';
+
+const paths = [
+  `app${all}${valid}`,
+  `build${all}${valid}`,
+  `shared${all}${valid}`,
+  `test${all}${valid}`,
+  `ui${all}${valid}`,
+];
+
+const ignored = [
+  // Build modules not imported anywhere, but used at the CLI.
+  'electron-rebuild',
+
+  // Test modules not imported anywhere, but used at the CLI.
+  'mocha',
+  'coveralls',
+  'nyc',
+
+  // Modules used only for configuration.
+  'babel-preset-es2015-node5',
+  'babel-preset-react',
+  'babel-preset-stage-2',
+  'babel-eslint',
+  'eslint',
+  'eslint-config-airbnb',
+  'eslint-plugin-flow-vars',
+  'eslint-plugin-import',
+  'eslint-plugin-react',
+];
+
+// We can afford being a little more relaxed with the regexes here
+// since we have very strict eslint rules about whitespace.
+const requires = /require\('([\w-/]+)'\)/g;
+const imports = /import(?:.*?from)?\s+'([\w-/]+)'/g;
+
+describe('packages', () => {
+  it('should not be orpahned', test(async function() {
+    const manifest = await fs.readJson('package.json');
+    const modules = [
+      ...Object.keys(manifest.dependencies),
+      ...Object.keys(manifest.devDependencies),
+    ];
+
+    const sources = await globMany(paths);
+    const matches = await regexFiles(sources, requires, imports);
+    const expected = Array.from(new Set([...matches, ...ignored]));
+
+    expect(difference(modules, expected)).toEqual([]);
+  }));
+});
+
+function test(runner) {
```

Comment:
It's a bit annoying having to try catch every `await` and calling done() myself with the error argument to make the test fail, or with no arguments to make it pass. This utility function handles that for me. I'd actually suggest putting it in shared so more people use it.